src/scrot.c

```diff
@@ -351,17 +351,25 @@ static long miliToNanoSec(int ms)
     return ms * 1000L * 1000L;
 }
 
-void scrotSleepFor(int ms)
+struct timespec clock_now(void)
+{
+    struct timespec ret;
+    clock_gettime(CONTINUOUS_CLOCK, &ret);
+    return ret;
+}
+
+struct timespec scrotSleepFor(struct timespec start, int ms)
 {
     scrotAssert(ms >= 0);
-    struct timespec tmp;
-    clock_gettime(CONTINUOUS_CLOCK, &tmp);
-    tmp.tv_sec  += ms / 1000;
-    tmp.tv_nsec += miliToNanoSec(ms % 1000);
-    bool overflow = tmp.tv_nsec >= miliToNanoSec(1000);
-    tmp.tv_sec  += overflow;
-    tmp.tv_nsec -= overflow ? miliToNanoSec(1000) : 0;
-    scrotSleepUntil(&tmp);
+    struct timespec end = {
+        .tv_sec  = (ms / 1000) + start.tv_sec,
+        .tv_nsec = miliToNanoSec(ms % 1000) + start.tv_nsec,
+    };
+    bool overflow = end.tv_nsec >= miliToNanoSec(1000);
+    end.tv_sec  += overflow;
+    end.tv_nsec -= overflow ? miliToNanoSec(1000) : 0;
+    scrotSleepUntil(&end);
```

Comment:
This would be the obvious code:
```c
if (end.tv_nsec >= milliToNanoSec(1000)) {
    end.tv_sec++;
    end.tv_nsec -= miliToNanoSec(1000);
}
```